src/EncompassRest/ResourceLocks/ResourceLocks.cs

```diff
@@ -0,0 +1,137 @@
+﻿using EncompassRest.Loans;
+using EncompassRest.Loans.Enums;
+using EncompassRest.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace EncompassRest.ResourceLocks
+{
+    internal sealed class ResourceLocks : ApiObject
+    {
+        internal ResourceLocks(EncompassRestClient client) : 
+            base(client, "encompass/v1/resourceLocks")
+        {
+        }
+
+        public Task<ResourceLock> GetResourceLockAsync(ResourceLock resourceLock, CancellationToken cancellationToken = default)
```

Comment:
Can we remove this `ResourceLock` input parameter override?

Reply:
i would like to leave it for now because if you are checking an existing resourcelock it may be easier to pass in the ResourceLock.  I have specific use cases for that.

Reply:
The `ResourceLock` has an `Id` property though which you can use instead. We're not doing this with any other API's, e.g. there is no `Loans.GetLoanAsync(Loan loan)` method.

Reply:
yes, but loan does not require 3 parameters to get it again.  it is not just `ResourceLock.Id`

I would leave it until EM decides how they will handle the update to this function.

Reply:
For now it doesn't bother me as the class is internal.

Reply:
opened issue #95 for this